system/jlib/jsocket.cpp

```diff
@@ -2986,17 +3019,12 @@ const char * GetCachedHostName()
             ip.getIpText(ips);
             if (ips.length())
             {
-                cachehostname.set(ips.str());
                 cachehostip.ipset(ip);
-                return cachehostname.get();
+                if (!resolveHostnames)
+                    cachehostname.set(ips.str());
             }
         }
 #endif
```

Comment:
cachehostname is the IP, if that can be figured out, otherwise its the hostname - this code is just  keeping it as it was.  myhostname is the hostname and does not get turned into an IP.